Zenith/Controllers/EventsController.cs

```diff
@@ -22,7 +24,57 @@ public class EventsController : Controller
             var events = db.Events.Include(E => E.ActivityCategory);
             return View(await events.ToListAsync());
         }
+        public async Task<ActionResult> Schedule()
+        {
+            return View(getDic());
+        }
+        public ScheduleViewModel getDic()
+        {
+            DateTime startOfWeek = DateTime.Today.AddDays(
+                (int)CultureInfo.CurrentCulture.DateTimeFormat.FirstDayOfWeek -
+                (int)DateTime.Today.DayOfWeek);
+
+            string result = string.Join("," + Environment.NewLine, Enumerable
+              .Range(0, 7)
+              .Select(i => startOfWeek
+                 .AddDays(i)
+                 .ToString("dd-MMMM-yyyy")));
+            var weekDays = result.Split(',');
+            List<DateTime> dates = new List<DateTime>();
+            foreach(var w in weekDays)
+            {
+                dates.Add(Convert.ToDateTime(w));
+            }
+
```

Comment:
The way you're making the dates of the week is too complicated. You can replace lines 37-47 in one line basically:

List<DateTime> list = Enumerable.Range(0, 7).Select(i => startOfWeek.AddDays(i)).ToList<DateTime>();